src/main/java/com/eternalcode/parcellockers/user/UserRepository.java

```diff
@@ -0,0 +1,17 @@
+package com.eternalcode.parcellockers.user;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+public interface UserRepository {
+
+    CompletableFuture<Void> save(User user);
+
+    Optional<User> findByName(String name);
+
+    Optional<User> findByUuid(UUID uuid);
+
+    List<User> findAll();
```

Comment:
możesz opakować w completabla

Reply:
Jest sens pakowania tego w `Completable`/`CompletableFuture` w przypadku gdy `JdbcConnectionProvider` wysyłania asynchronicznie zapytania?

Reply:
> Jest sens pakowania tego w `Completable`/`CompletableFuture` w przypadku gdy `JdbcConnectionProvider` wysyłania asynchronicznie zapytania?

Tego chyba nie wysyła async jak dobrze widziałem :/, ale przydałoby się wybrać jedną z opcji.

Reply:
https://github.com/EternalCodeTeam/ParcelLockers/blob/feature/database/src/main/java/com/eternalcode/parcellockers/database/JdbcConnectionProvider.java Tutaj jest używane CompletableFuture#supplyAsync, czy to nie wystarczy?

Reply:
Dobra fixed